### PR TITLE
feat(optimus): [RND-103047] Add Notification component 

### PR DESCRIPTION
**Pull request by witwash**
#### Summary

This PR adds a Notification component. Showing notifications and stacking them will be added in separate PR (RND-104514).

#### Testing steps

None

#### Follow-up issues

RND-104514 with implemented behavior. 

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).

### CODE REVIEW

File: storybook/lib/stories/notification.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      title: title,
+      body: body.isNotEmpty ? body : null,
+      variant: variant,
+      link: link.isNotEmpty ? link : null,
+      onDismissed: dismissible ? () {} : null,
+    );
+  },
+);
+
+final _variants = OptimusNotificationVariant.values.toOptions();
```
Comment (ookami-kb) [inline storybook/lib/stories/notification.dart:30]: nitpick: I would inline it.

File: optimus/lib/src/notification/notification.dart [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+  final String title;
+  final String? body;
+  final IconData? icon;
+  final String? link;
+  final VoidCallback? onLinkPressed;
+  final VoidCallback? onDismissed;
+  final OptimusNotificationVariant variant;
+
+  double _padding(BuildContext context) {
```
Comment (ookami-kb) [inline optimus/lib/src/notification/notification.dart:49]: Nitpicking: we rather use `_getPadding()` for methods and `padding` for properties.

File: optimus/lib/src/notification/notification.dart [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+  }
+}
+
+class _NotificationCloseButton extends StatelessWidget {
+  const _NotificationCloseButton({
+    Key? key,
+    required this.onDismissed,
+  }) : super(key: key);
+
+  final VoidCallback? onDismissed;
```
Comment (ookami-kb) [inline optimus/lib/src/notification/notification.dart:302]: For button, it doesn't make sense to have nullable `onDismissed`. If the button is there, callback shoud be present.

File: optimus/lib/src/notification/notification.dart [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+            _NotificationContent(
+              icon: icon,
+              variant: variant,
+              title: title,
+              body: body,
+              link: link,
+              onLinkPressed: onLinkPressed,
+            ),
+            if (onDismissed != null)
+              _NotificationCloseButton(onDismissed: onDismissed)
```
Comment (ookami-kb) [inline optimus/lib/src/notification/notification.dart:84]: It can cause an overlap:

image

**Review by ookami-kb — CHANGES_REQUESTED** [review]
Some minor comments, otherwise LGTM.

File: storybook/lib/stories/notification.dart (hunk 1 above)
Comment (witwash) [reply storybook/lib/stories/notification.dart]: Hope you meant this.
Screenshot 2022-05-25 at 16 16 47

**Pull request merged**